security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/JwtOAuth2Service.scala

```diff
@@ -49,8 +51,9 @@ class JwtOAuth2Service[
       Future(accessToken)
         .flatMap(accessToken => introspectJwtToken[AccessTokenClaims](accessToken))
         .flatMap(claims =>
-          if (requiredAccessTokenAudience.isEmpty || claims.audienceAsList.exists(requiredAccessTokenAudience.contains))
-            Future.successful(claims.expirationTime)
+          if (requiredAccessTokenAudience.isEmpty || claims.audienceAsList.exists(requiredAccessTokenAudience.contains)) {
+            Future.successful(claims.expirationTime.map(_.toEpochMilli).map(t => Deadline(FiniteDuration(t - System.currentTimeMillis(), TimeUnit.MILLISECONDS))))
```

Comment:
:+1:  Let's return Instant here, and extract a method/class and make it along the lines:
```
    val validDuration = scala.concurrent.duration.Duration.fromNanos(java.time.Duration.between(Instant.now(), Instant.now()).toNanos) ==> maybe it can be nicer :) 
    Deadline.now + validDuration => this should be in checkAuthorizationAndObtainUserinfo
```

Reply:
the reson it's done exactly here and with deadline is that in every place further in code we will loose (wrongly compute) expiration period due to passage of time